hive-plugins/src/main/java/co/cask/plugin/etl/batch/commons/HiveSchemaConverter.java

```diff
@@ -49,12 +49,14 @@
    */
   public static HCatSchema toHiveSchema(Schema schema, HCatSchema tableSchema) {
     List<HCatFieldSchema> fields = Lists.newArrayList();
+    //TODO: Finalize a consistent way to whether do projecttion on source or not while reading and if needed
```

Comment:
Let's say Hive tables has 5 fields A,B,C,D & E. If the hive source output schema contains only B,D & E then the record has to be projected.